src/erleans_partitions.erl

```diff
@@ -0,0 +1,164 @@
+    partisan_peer_service_events:add_sup_callback(fun(Membership) ->
+                                                      Self ! {update, Membership}
+                                                  end),
+    NumPartitions = erleans_config:get(num_partitions),
+    {ok, #state{num_partitions=NumPartitions,
+                node_ranges=[],
+                to_notify=#{}}, 0}.
+
+handle_call({find_node, Item}, From, State=#state{num_partitions=NumPartitions,
+                                                  node_ranges=Ranges}) ->
+    spawn(fun() ->
+              Partition = ?CH(erlang:phash2(Item), NumPartitions),
+              {_, Node} = ec_lists:fetch(fun({{Start, End}, _Node}) when Partition >= Start
+                                                                       , Partition =< End ->
+                                             true;
+                                            (_) ->
+                                             false
+                                         end, Ranges),
+              gen_server:reply(From, {Partition, Node})
+          end),
+    {noreply, State};
+handle_call(get_range, _From, State=#state{range=Range}) ->
+    {reply, Range, State};
+handle_call({add_handler, Name, Pid}, _From, State=#state{to_notify=ToNotify}) ->
+    {reply, ok, State#state{to_notify=ToNotify#{Name => Pid}}}.
+
+handle_cast(_Msg, State) ->
+    {noreply, State}.
+
+handle_info({update, Membership}, State=#state{num_partitions=NumPartitions,
+                                               to_notify=ToNotify}) ->
+    MembersList = lists:usort(sets:to_list(state_orset:query(Membership))),
+    {Range, NodeRanges} = update_ranges(MembersList, NumPartitions, ToNotify),
+    {noreply, State#state{range=Range,
+                          node_ranges=NodeRanges}};
+handle_info(timeout, State=#state{num_partitions=NumPartitions,
+                                  to_notify=ToNotify}) ->
+    {ok, MembersList} = partisan_peer_service:members(),
+    {Range, NodeRanges} = update_ranges(MembersList, NumPartitions, ToNotify),
+    {noreply, State#state{range=Range,
+                          node_ranges=NodeRanges}}.
+
+code_change(_OldVsn, State, _Extra) ->
+    {ok, State}.
+
+terminate(_Reason, _State) ->
+    ok.
+
+%% Internal functions
+
+update_ranges(MembersList, NumPartitions, ToNotify) ->
+    Length = length(MembersList),
+    {_, NodeRanges} = lists:foldl(fun({Node, _, _}, {Pos, Acc}) ->
+                                      Range = calc_partition_range(Pos, Length, NumPartitions),
+                                      {Pos+1, [{Range, Node} | Acc]};
+                                     (Node, {Pos, Acc}) ->
+                                      Range = calc_partition_range(Pos, Length, NumPartitions),
+                                      {Pos+1, [{Range, Node} | Acc]}
+                                  end, {0, []}, MembersList),
+
+    maps:map(fun(_, Pid) ->
+                 Pid ! update_streams
```

Comment:
can't it calculate ranges here and spare the receiving pid the trouble of having to make another `call` to this process?